File: 70-climbing-stairs/70-climbing-stairs.js
/**
 * @param {number} n
 * @return {number}
 */
var climbStairs = function(n) {
    let answers = [null,1,2];
    
    for(let i = 3; i <= n; i++){
        answers[i] = answers[i-1] + answers[i-2];
    }
    return answers[n];
};

File: 70-climbing-stairs/70-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        dp=[0]*(n+2)
        dp[0]=1
        for i in range(len(dp)-2):
            dp[i+1]+=dp[i]
            dp[i+2]+=dp[i]
            
        return dp[n]
        